hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -3193,6 +3266,30 @@ public String getCanonicalServiceName() {
   private CopyResult copyFile(String srcKey, String dstKey, long size,
       S3ObjectAttributes srcAttributes, S3AReadOpContext readContext)
       throws IOException, InterruptedIOException  {
+    return copyFile(bucket, srcKey, bucket, dstKey, size, srcAttributes, readContext);
```

Comment:
this doesn't update S3Guard BTW. IF you want newly created files to be visible, your copy operation owns this problem